hangups/http_utils.py

```diff
@@ -4,61 +4,170 @@
 import asyncio
 import collections
 import logging
-import os
+import urllib.parse
 
-from hangups import exceptions
+from hangups import exceptions, channel
 
 logger = logging.getLogger(__name__)
 CONNECT_TIMEOUT = 30
 REQUEST_TIMEOUT = 30
 MAX_RETRIES = 3
 
-FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body',
-                                                         'cookies'])
+FetchResponse = collections.namedtuple('FetchResponse', ['code', 'body'])
 
 
-@asyncio.coroutine
-def fetch(session, method, url, params=None, headers=None, data=None):
-    """Make an HTTP request.
+class ClientSession(aiohttp.ClientSession):
```

Comment:
Would it be easy enough to have `ClientSession` as a private attribute now instead of inheriting from it?

Reply:
But keep the name `http_utils.ClientSession`? This might be a bit confusing.

Reply:
At that point we could name it to something different like `Session`.

Reply:
Changed in https://github.com/tdryer/hangups/pull/342/commits/3690f6fed2258b40f4d703bc31dd59416df6e10f